src/DebuggingFeature.ts

```diff
@@ -0,0 +1,80 @@
+import { BaseLanguageClient, StaticFeature, Neovim } from "coc.nvim";
+import {
+  ExecuteCommandParams,
+  ExecuteCommandRequest,
+  InitializeParams,
+  ServerCapabilities
+} from "vscode-languageserver-protocol";
+import { commands } from "coc.nvim";
+
+interface DebugAdapterStartResponse {
+  name: string;
+  uri: string;
+}
+
+export class DebuggingFeature implements StaticFeature {
+  private withVimspector = false;
+
+  constructor(private _nvim: Neovim, private _client: BaseLanguageClient) {}
+
+  async preInit(): Promise<void> {
+    this.withVimspector = (await this._nvim.getVar("loaded_vimpector")) === 1;
+  }
+
+  fillInitializeParams(params: InitializeParams): void {
+    if (this.withVimspector) {
+      if (params.capabilities.experimental == null) {
+        params.capabilities.experimental = {};
+      }
+      params.capabilities.experimental.debuggingProvider = true;
```

Comment:
Currently with Metals starting the process by being triggered from a code lens, this means the code lens needs to be present. This is working with Nvim, but not with Vim. So I think until we solve that issue, we'll need to do a check prior to any of this to see if the user is using Vim or not before we can actually decide if we should turn on this feature.

Reply:
Fixed.

Reply:
```suggestion
      (params.capabilities.experimental as any).debuggingProvider = true;
```

Reply:
Done.